pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -495,7 +495,7 @@ func BuildSidecarOutboundVirtualHosts(node *model.Proxy, push *model.PushContext
 func dedupeDomains(domains []string, vhdomains sets.String, expandedHosts []string, knownFQDNs sets.String) []string {
 	temp := domains[:0]
 	for _, d := range domains {
-		if vhdomains.Contains(d) {
+		if vhdomains.Contains(strings.ToLower(d)) {
```

Comment:
i think we need to convert for `vhdomains.Insert(d)`. otherwise we can still return  domains only diff in case

Reply:
Is doing validation in webhook better than internally handling here? As users may potentially confused about whether the field is case sensitive or not. However the validation may have some compatibility issues though.

Reply:
vhdomains.Insert(d) - Done

Doing validation in webhook is not possible across virtual services

Reply:
I mean reject the upper case, currently our validation accepts upper case as well.

Reply:
But we can not enforce stricter validations now though

Reply:
Yes that's my concern as well. Thanks

Reply:
we can add a warning though

Reply:
https://github.com/istio/istio/pull/45725#pullrequestreview-1504985265 - based on this we can not add warning?

Reply:
TBH that issue did not make sense to me. It basically says "DNS is case insensitive, so we should allow any case". That is wrong - its insensitive so we should enforce the canonical form (lowercase) IMO. `host: eXaMpLe.com` is not a good idea. We cannot reject now unfortunately but we can still warn?

Reply:
We can still warn - We can just say "host does not follow canonical DNS naming and may create problems in routing" ?